docs/experiments/data-pipeline.md

```diff
@@ -0,0 +1,51 @@
+---
+sidebar_position: 14
+---
+
+# Data pipeline
+
+At a high level, the Eppo experiment data pipeline has four main branches along with an assignment summarization that is a predecessor for each of those branches.
+
+1. Assignment summarization
+2. Core pipeline
+3. CUPED pipeline
+4. Metric dimensions/properties pipeline
+5. Funnel metrics pipeline
+
+![Pipeline visualization][/img/experiments/data-pipeline/experiment_pipeline_visualization.png]
+
+## Assignment summarization
+
+1. **Incremental daily assignments**: As part of the incremental process, we first delete the last two days worth of data. We then select all assignments from the past two days for this experiment and insert them into a daily table, recording the first variant, assigned timestamp, and whether or not a subject was exposed to more than one variant for each day.
+2. **Assignments summary**: We recompute the assignment summary from scratch each run. There's one record for every subject, including the observed variant, timestamp of assignment, and whether or not they were exposed to multiple variants. This table is used downstream by all four pipeline branches.
+
+## Core Pipeline
+
+The core pipeline starts with the assignments, incrementally joins event data from your metrics/facts, aggregates the data, and computes winsorized aggregates.
+
+1. **Incremental daily data frame**: We join your experiment assignments to your fact sources. We use a "wide" table format where every metric aggregation is a separate column. As part of the incremental process, we first delete two days of data before re-inserting it. This is to ensure if there's time-delayed data it's updated appropriately.
+2. **Cumulative data frame**: We aggregate the daily data to generate a timeline view of all metric data for each subject.
+3. **Winsorized data frame**: We apply winsorization according to the cumulative data frame based on your metric settings.
+4. **Dimensionalized summaries**: We convert our subject-level winsorized data frame to an overall summary with dimensional values included.
+5. **Dimensionalized summaries reduced**: For high cardinality dimensions, we cap dimension values to 50 to minimize the data we transfer to our operational Postgres database.
```

Comment:
How are the 50 chosen? The 50 most frequent?

Reply:
By assignment volume.  Will clarify.